config/cylc_hpc/flow.cylc

```diff
@@ -106,82 +102,89 @@
     
     [[pulljuliaimage]]
         script = """
-            apptainer pull --force $project_dir/icefloetracker-julia.simg docker://brownccv/icefloetracker-julia:pr-45
+            apptainer pull --force $project_dir/icefloetracker-julia.simg docker://brownccv/icefloetracker-julia:main
         """
         platform = oscar
         execution time limit = PT1H
         submission retry delays = 2*PT30S
         execution retry delays = 2*PT6S
         [[[directives]]]
-            --mem = 16G
+            --mem = 32G
             --cpus-per-task = 8
     
-    [[landmask<job_id>]]
+    [[landmask<param_set>]]
         script = """
-            i=${CYLC_TASK_PARAM_job_id}
+            i=${CYLC_TASK_PARAM_param_set}
+            location_array=( $(echo $location | sed -e 's/,/ /g') )
 
-            singularity exec --bind $landmask_dir/$i:/tmp,$report_dir:/usr/local/bin/../report $project_dir/icefloetracker-julia.simg $julia_exec -t auto /usr/local/bin/ice-floe-tracker.jl landmask $fetchdata_dir/$i /tmp
+            singularity exec --bind $results_dir/${location_array[$i]}/landmasks:/tmp,$report_dir:/usr/local/bin/../report $project_dir/icefloetracker-julia.simg $julia_exec -t auto /usr/local/bin/ice-floe-tracker.jl landmask $fetchdata_dir/${location_array[$i]} /tmp
         """
         platform = oscar
         execution time limit = PT1H
         submission retry delays = 2*PT30S
-        execution retry delays = 2*PT6S
+        execution retry delays = PT6S
         [[[directives]]]
             --mem = 20G
             --cpus-per-task = 16
     
-    [[preprocess<job_id>]]
+    [[preprocess<param_set>]]
         script = """
-            i=${CYLC_TASK_PARAM_job_id}
+            i=${CYLC_TASK_PARAM_param_set}
+            location_array=( $(echo $location | sed -e 's/,/ /g') )
 
-            singularity exec --bind $preprocess_dir/$i:/tmp,$report_dir:/usr/local/bin/../report $project_dir/icefloetracker-julia.simg $julia_exec -t auto /usr/local/bin/ice-floe-tracker.jl preprocess -t $fetchdata_dir/$i/truecolor -r $fetchdata_dir/$i/reflectance -l $landmask_dir/$i -p $soit_dir/$i -o /tmp
+            singularity exec --bind $results_dir/${location_array[$i]}/preprocess:/tmp,$report_dir:/usr/local/bin/../report $project_dir/icefloetracker-julia.simg $julia_exec -t auto /usr/local/bin/ice-floe-tracker.jl preprocess -t $fetchdata_dir/${location_array[$i]}/truecolor -r $fetchdata_dir/${location_array[$i]}/reflectance -l $results_dir/${location_array[$i]}/landmasks -p $results_dir/${location_array[$i]}/soit -o /tmp
         """
         platform = oscar
         execution time limit = PT1H
         submission retry delays = 2*PT30S
-        execution retry delays = 2*PT6S
+        execution retry delays = PT6S
         [[[directives]]]
             --mem = 60G
             --cpus-per-task = 40
```

Comment:
Make equal to number of images?

Reply:
see issue #82. I tried counting the number of images in the fetchdata/truecolor dir and also calculating days between end and start dates. Neither seemed to work properly.